apps/heft/src/plugins/JestPlugin/JestTypeScriptDataFile.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { JsonFile } from '@rushstack/node-core-library';
+import { ITypeScriptConfiguration } from '../../stages/BuildStage';
+
+/**
+ * Schema for jest-typescript-data.json
+ */
+export interface IJestTypeScriptDataFileJson {
+  /**
+   * The "emitFolderPathForJest" from .heft/typescript.json
+   */
+  emitFolderPathForJest: string;
+}
+
+/**
+ * Manages loading/saving the "jest-typescript-data.json" data file.  This file communicates
+ * configuration information from Heft to jest-build-transform.js.  The jest-build-transform.js script gets
+ * loaded dynamically by the Jest engine, so it does not have access to the normal HeftConfiguration objects.
+ */
+export class JestTypeScriptDataFile {
+  /**
+   * Called by TypeScriptPlugin to write the file.
+   */
+  public static saveForProject(
+    projectFolder: string,
+    typeScriptConfiguration: ITypeScriptConfiguration
+  ): void {
+    const jsonFilePath: string = JestTypeScriptDataFile.getConfigFilePath(projectFolder);
+
+    const json: IJestTypeScriptDataFileJson = {
+      emitFolderPathForJest: typeScriptConfiguration.emitFolderPathForJest || 'lib'
```

Comment:
Think we should log something if this isn't defined?

Reply:
We document that the default value is `lib`